### DISCUSSION TITLE
.s as prompt in Zeptoforth

### DISCUSSION THREAD

**Discussion by robzed** [tabemann/zeptoforth#62] [Ideas]
Flashforth has the nice feature of showing the parameter stack view on every prompt.

It also shows what base it's in and whether it's compiling to Flash or RAM - like ok<#, RAM> or ok<$, RAM>

What's the easiest way to this on Zeptoforth?

**Comment by tabemann** [comment]
You can accomplish this via something like the following:

```
: type-my-base ( -- )
  base @ dup { saved-base } case
    #10 of ." #" endof
    #16 of ." $" endof
    #8 of ." /" endof
    #2 of ." %" endof
    dup #10 base ! (.) saved-base base !
  endcase
;

: my-prompt ( ? -- ? )
  ." ok<"
  type-my-base
  compiling-to-flash? if ." , flash>" else ." , RAM>" then
  .s
  cr
;

' my-prompt prompt-hook !
```

**Reply by robzed** [reply]
This is great ... will try it out later.

**Reply by robzed** [reply]
This works for me (Zeptoforth 1.0.2)

**Reply by robzed** [reply]
Also on 1.0.3.1

 ok<#, RAM> [ ]
1 ok<#, RAM> [ 1 ]
3 ok<#, RAM> [ 1 3 ]
10 ok<#, RAM> [ 1 3 10 ]
100 ok<#, RAM> [ 1 3 10 100 ]
hex ok<$, RAM> [ 1 3 A 64 ]
decimal ok<#, RAM> [ 1 3 10 100 ]

**Comment by tabemann** [comment]
Note that this will break uploads with `codeload3.py` /`codeload3.sh` because those expect the line ending `ok`\<CR>\<LN>.

**Comment by tabemann** [comment]
However, it will be compatible with zeptocom.js because zeptocom.js listens for ACK for flow control, which is separate from the prompt hook.

**Reply by robzed** [reply]
I've actually swapped from codeload3 to zeptocom.js ... works well, and so that's in my current work flow.

**Comment by tabemann** [comment]
One important note is that there are many places where data is left on the data stack, such as when defining anonymous words, flow control constructs, structures, and classes. So don't be alarmed when you see extra values on the stack in these contexts.
